### ISSUE TITLE
Bundle COB_CONFIG_DIR

### ISSUE THREAD

**Issue by bclement-ocp** [OCamlPro/superbol-studio-oss#90]
So that dialects work.

**Comment by GitMensch** [comment]
If this is done then - on a "native" environment there should _first_ be an invocation of `cobc --info` to check for the system installed/configured `COB_CONFIG_DIR` and only if this does not exist fall back to the bundled version.

**Comment by bclement-ocp** [comment]
Agreed — this is intended as a last-resort fallback if no `cobc` is installed. The order of preference should be:

1. The value of the `COB_CONFIG_DIR` environment variable if defined; or
2. `COB_CONFIG_DIR` from `cobc --info`, if there is an installed `cobc`; or
3. The bundled version

This would allow features that use the config (anything related to dialects, mostly) to work even without a working GNUCobol installation.

**Comment by GitMensch** [comment]
Right - good plan.

Note: You don't _need_ to do 1 (commonly only set with dev environments or Win32) as 2 will also show that value, but 1 will be an unexpensive function call while 2 is a subprocess, so it still may be reasonable (also because you don't have to parse the environment variant from the output).

> we currently only do 1. with a hardcoded default value of `/usr/local/share/gnucobol/config`

OK, that's obviously bad, because that won't get system-installed GC versions and also won't work with other filesystem layouts. Your suggested order fixes that and is likely not that hard to implement.
